examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,336 @@
+"""
+Plotting focal mechanisms
+=========================
+
+Focal mechanisms can be plotted as beachballs with the :meth:`pygmt.Figure.meca`
+method.
+
+The focal mechanism data or parameters can be provided as different input types:
+external file, 1-D or 2-D ``numpy.array``, dictionary, ``pandas.Dataframe``.
+Different conventions to define the focal mechanism are supported: Aki and Richards
+(``"aki"``), global CMT (``"gcmt"``), moment tensor (``"mt"``), partial focal
+mechanism (``"partial"``), principal axis (``"principal_axis"``). Please refer to
+the documentation of :meth:`pygmt.Figure.meca` regarding how to set up the input
+data in respect to the chosen input type and convention (i.e the expected column
+order, keys, or column names).
+
+This tutorial focus on how to adjust the display of the beachballs:
+
+- Filling the quadrants
+- Plotting the components of a seismic moment tensor
+- Adjusting the outlines
+- Highlighting the nodal planes
+- Adding offset from the event location
+- Adding a label
+- Using size-coding and color-coding
+"""
+
+# %%
+import pandas as pd
+import pygmt
+
+# Set up arguments for basemap
+size = 5
+projection = "X10c/4c"
+frame = ["af", "+ggray90"]
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
+
+# moment tensor convention
```

Comment:
Any specific reason that you need to define two focal mechanims here?

Reply:
Hm. I think I wanted to have the moment tensor convention for section "Plotting the components of a seismic moment tensor" (around line 109) and later the Aki and Richards convention for the sections "Adjusting the outlines" (around line 143) and "Highlighting the nodal planes" (around line 173).

Reply:
Hm. I think I wanted to have the moment tensor convention for section "Plotting the components of a seismic moment tensor" (around line 109) and later the Aki and Richards convention for the sections "Adjusting the outlines" (around line 143) and "Highlighting the nodal planes" (around line 173).

Reply:
OK, I see your points. What about renaming them to shorter names like `meca_aki`/`meca_mt`?

Reply:
My idea of the variable names was ``convention_datatype_amount``:
- aki_dict_single
- mt_dict_single
- aki_dict_multiple
- aki_df_multiple